crates/oci-distribution/src/client.rs

```diff
@@ -402,11 +402,27 @@ impl Client {
         Ok(())
     }
 
+    /// Pull a manifest from the remote OCI Distribution service.
+    ///
+    /// The client will check if it's already been authenticated and if
+    /// not will attempt to do.
+    pub async fn pull_manifest(
+        &mut self,
+        image: &Reference,
+        auth: &RegistryAuth,
+    ) -> anyhow::Result<(OciManifest, String)> {
+        if !self.tokens.contains_key(image.registry()) {
+            self.auth(image, auth, &RegistryOperation::Pull).await?;
+        }
+
+        self._pull_manifest(image).await
+    }
+
     /// Pull a manifest from the remote OCI Distribution service.
     ///
     /// If the connection has already gone through authentication, this will
     /// use the bearer token. Otherwise, this will attempt an anonymous pull.
-    async fn pull_manifest(&self, image: &Reference) -> anyhow::Result<(OciManifest, String)> {
+    async fn _pull_manifest(&self, image: &Reference) -> anyhow::Result<(OciManifest, String)> {
```

Comment:
Not sure about the convention to prepend unexported functions with `_`, but otherwise, the change makes sense.

Reply:
I've seen that in several other rust crates FWIW